mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -1124,8 +1124,11 @@ class SubscriptIDOffset : public AccessorSubscriptIDIndexPattern {
   using AccessorSubscriptIDIndexPattern::AccessorSubscriptIDIndexPattern;
 
   LogicalResult match(SYCLAccessorSubscriptOp op) const final {
-    return success(AccessorSubscriptPattern::hasIDOffsetType(op) &&
-                   op.getType().isa<MemRefType>());
```

Comment:
In the added test case, `op` is `%0 = sycl.accessor.subscript %acc[%idx] { ArgumentTypes = [memref<?x!sycl_accessor_1_id_rw_gb>, memref<?x!sycl_id_1_>], FunctionName = @"operator[]", MangledFunctionName = @"operator[]", TypeName = @"id" }  : (memref<?x!sycl_accessor_1_id_rw_gb>, memref<?x!sycl_id_1_>) -> memref<?x!sycl_id_1_, 1>`, so `op.getType()` is `memref<?x!sycl_id_1_, 1>`, which is a `MemRefType`. Shouldn't the original `match` function already match it?

Reply:
You're right. Pushed wrong test. Fixed!